src/game/logic/object/weapon.cpp

```diff
@@ -155,6 +148,172 @@ Weapon *WeaponStore::Allocate_New_Weapon(const WeaponTemplate *tmpl, WeaponSlotT
+        WeaponTemplate *tmplate = m_weaponTemplateVector[i];
+
+        if (tmplate != nullptr) {
+            tmplate->Delete_Instance();
+        }
+    }
+
+    m_weaponTemplateVector.clear();
+}
+
+void WeaponStore::Handle_Projectile_Detonation(const WeaponTemplate *tmplate,
+    const Object *source_obj,
+    const Coord3D *victim_pos,
+    int bonus_condition,
+    bool do_damage) const
+{
+    Weapon *weapon = g_theWeaponStore->Allocate_New_Weapon(tmplate, WEAPONSLOT_PRIMARY);
+    weapon->Load_Ammo_Now(source_obj);
+    weapon->Fire_Projectile_Detonation_Weapon(source_obj, victim_pos, bonus_condition, do_damage);
+    weapon->Delete_Instance();
+}
+
+void WeaponStore::Create_And_Fire_Temp_Weapon(
+    const WeaponTemplate *tmplate, const Object *source_obj, const Coord3D *victim_pos)
+{
+    if (tmplate) {
+        Weapon *weapon = g_theWeaponStore->Allocate_New_Weapon(tmplate, WEAPONSLOT_PRIMARY);
+        weapon->Load_Ammo_Now(source_obj);
+        weapon->Fire_Weapon(source_obj, victim_pos, nullptr);
+        weapon->Delete_Instance();
+    }
+}
+
+WeaponTemplate *WeaponStore::Find_Weapon_Template_Private(NameKeyType key) const
+{
+    for (unsigned int i = 0; i < m_weaponTemplateVector.size(); i++) {
+        WeaponTemplate *tmplate = m_weaponTemplateVector[i];
+
+        if (tmplate->Get_Name_Key() == key) {
+            return tmplate;
+        }
+    }
+
+    return nullptr;
+}
+
+WeaponTemplate *WeaponStore::New_Weapon_Template(Utf8String name)
+{
+    if (name.Is_Empty()) {
+        return nullptr;
+    } else {
+        WeaponTemplate *tmplate = new WeaponTemplate();
+        tmplate->m_name = name;
+        tmplate->m_nameKey = g_theNameKeyGenerator->Name_To_Key(name.Str());
+        m_weaponTemplateVector.push_back(tmplate);
+        return tmplate;
+    }
+}
+
+WeaponTemplate *WeaponStore::New_Override(WeaponTemplate *tmplate)
+{
+    if (tmplate == nullptr) {
+        return nullptr;
+    }
+
+    WeaponTemplate *t = new WeaponTemplate();
+    t = tmplate;
+    t->Friend_Set_Next_Template(tmplate);
+    return t;
+}
+
+void WeaponStore::Update()
+{
+    for (auto i = m_weaponDDI.begin(); i != m_weaponDDI.end();) {
+        if (g_theGameLogic->Get_Frame() < i->m_delayDamageFrame) {
+            i++;
+        } else {
+            i->m_delayedWeapon->Deal_Damage_Internal(
+                i->m_delaySourceID, i->m_delayIntendedVictimID, &i->m_delayDamagePos, i->m_bonus, false);
+            i = m_weaponDDI.erase(i);
+        }
+    }
+}
+
+void WeaponStore::Delete_All_Delayed_Damage()
+{
+    m_weaponDDI.clear();
+}
+
+void WeaponStore::Reset_Weapon_Templates()
+{
+    for (unsigned int i = 0; i < m_weaponTemplateVector.size(); i++) {
+        m_weaponTemplateVector[i]->Reset();
+    }
+}
+
+void WeaponStore::Reset()
+{
+    for (unsigned int i = 0; i < m_weaponTemplateVector.size(); i++) {
+        WeaponTemplate *tmplate = m_weaponTemplateVector[i];
+
+        if (tmplate->Get_Next_Template()) {
```

Comment:
should actually be a isOverride call 006DB780 wb
Friend_Clear_Next_Template returns what it popped, return isn't used in this function but i noticed its void

Reply:
fixed.